MBBSEmu.Tests/ExportedModules/Majorbbs/FileTestBase.cs

```diff
@@ -76,6 +76,54 @@ protected ushort fwrite(IntPtr16 destPtr, ushort size, ushort count, IntPtr16 fi
             return mbbsEmuCpuRegisters.AX;
         }
 
+        protected ushort f_printf(IntPtr16 filep, string formatString, params object[] values)
+        {
+            var fprintfParameters = new List<ushort> {filep.Offset, filep.Segment};
+
+            //Add Formatted String
+            var inputStingParameterPointer = mbbsEmuMemoryCore.AllocateVariable(Guid.NewGuid().ToString(), (ushort)(formatString.Length + 1));
+            mbbsEmuMemoryCore.SetArray(inputStingParameterPointer, Encoding.ASCII.GetBytes(formatString));
+            fprintfParameters.Add(inputStingParameterPointer.Offset);
+            fprintfParameters.Add(inputStingParameterPointer.Segment);
+
+            //Add Parameters
+            foreach (var v in values)
+            {
+                switch (v)
+                {
+                    case string @parameterString:
+                    {
+                        var stringParameterPointer = mbbsEmuMemoryCore.AllocateVariable(Guid.NewGuid().ToString(), (ushort)(@parameterString.Length + 1));
+                        mbbsEmuMemoryCore.SetArray(stringParameterPointer, Encoding.ASCII.GetBytes(@parameterString));
+                        fprintfParameters.Add(stringParameterPointer.Offset);
+                        fprintfParameters.Add(stringParameterPointer.Segment);
+                        break;
+                    }
+                    case uint @parameterULong:
+                    {
+                        var longBytes = BitConverter.GetBytes(@parameterULong);
+                        fprintfParameters.Add(BitConverter.ToUInt16(longBytes, 0));
+                        fprintfParameters.Add(BitConverter.ToUInt16(longBytes, 2));
+                        break;
+                    }
+                    case int @parameterLong:
+                    {
+                        var longBytes = BitConverter.GetBytes(@parameterLong);
+                        fprintfParameters.Add(BitConverter.ToUInt16(longBytes, 0));
+                        fprintfParameters.Add(BitConverter.ToUInt16(longBytes, 2));
+                        break;
+                    }
+                    case ushort @parameterInt:
```

Comment:
`@parameterUInt`

Also, add the `short` case just to cover all the bases?

Reply:
I pulled this from the `prf` tests. There might be a way to refactor this into the base as to not have multiple copies of the logic. Let me see what we can do there.

Reply:
To MBBSEMu, they're longs ;)